provisioner/data.js

```diff
@@ -1,30 +1,704 @@
-    // them... Or if you want to add a lot of JSON, it can be necessary to split
-    // it...
-  ]
+  version: 1,
+  partitionKey: base.Entity.keys.ConstantKey(KEY_CONST),
+  rowKey: base.Entity.keys.StringKey('workerType'),
+  properties: {
+    /* This is a string identifier of the Worker Type.  It
+     * is what we give to the Queue to figure out whether there
+     * is pending work. */
+    workerType: base.Entity.types.String,
+    /* This is the basic AWS LaunchSpecification.  It is
+     * stored as an opaque JSON blob and represents the
+     * information which will be shared between all instances
+     * of this worker across all regions and instance types */
+    launchSpecification: base.Entity.types.JSON,
+    /* This is the minimum capacity we will run */
+    minCapacity: base.Entity.types.Number,
+    /* This is the maximum capacity which we will ever run */
+    maxCapacity: base.Entity.types.Number,
+    /* Scaling ratio a ratio of pending jobs to capacity.  A number
+     * which is between 0 and 1 will ensure that there is always 
+     * idle capacity and a number greater than 1 will ensure that
+     * some percentage of tasks will remain pending before we spawn
+     * instances. */
+    scalingRatio: base.Entity.types.Number,
+    /* This is the minimum spot bid... It isn't actually read
+     * and should be moved to the instance type definition... */
+    minSpotBid: base.Entity.types.Number,
+    maxSpotBid: base.Entity.types.Number,
+    /* Right now this is ineffective */
+    canUseOndemand: base.Entity.types.JSON,
+    /* Right now this is ineffective */
+    canUseSpot: base.Entity.types.JSON,
+    /* This dictionary describes which instance types that this workerType
+     * can run on as well as type-specific information.
+     * This is a dictionary in the shape:
+     * {
+     *   'c1.small': {
+     *     capacity: 1,
+     *     utility: 1,
+     *     overwrites: {}
+     *   }
+     * }
+     * The top level key is the EC2 instance type which should
+     * be used.  In each instance type, there are three keys:
+     *   - capacity: this is the number of tasks this instance
+     *               can run concurrently.
+     *   - utility: this is an arbitrary number which we multiply
+     *              by spot price and compare to other instance
+     *              types to figure out which machine to bid on
+     *   - overwrites: this object overwrites keys in the general
+     *                 launch specification
+     */
+    types: base.Entity.types.JSON,
+    /* This is a JSON object which contains the regions which this
+     * workerType is allowed to run in as well as the region-specific
+     * information.  It is in the shape:
+     * {
+     *   'us-west-1': {
+     *     'overwrites': {
+     *       'ImageId': 'ami-aaaaaaa'
+     *      }
+     *   }
+     * } */
+    regions: base.Entity.types.JSON,
+  },
+  context: ['ec2', 'keyPrefix', 'pubKey', 'influx'],
```

Comment:
You problably don't need `influx` it might be useful with a `reporter` see `base.stats` on how to create a reporter when you have a stats drain like an instance of `Influx`.